File: Integration Examples/eAPI v1.8/java/eAPI-v18-connector/src/main/java/cz/monetplus/mips/eapi/v18/connector/entity/PayRefundReq.java
package cz.monetplus.mips.eapi.v18.connector.entity;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class PayRefundReq extends PayReq {

	private static final long serialVersionUID = -3825192932302805075L;
	
	@XmlElement(nillable=true)
	public Long amount;

	public PayRefundReq() {
		
	}

	public PayRefundReq(String merchantId, String payId, Long amount) {
		super(merchantId, payId);
		this.amount = amount;
	}

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, merchantId);
		add(sb, payId);
		add(sb, dttm);
		add(sb, amount);
		deleteLast(sb);
		return sb.toString();
	}

}

File: examples/eAPI v1.8/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/PayRefundReq.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects
{
    public class PayRefundReq : PayReq
    {
        [JsonProperty("amount")]
        public long Amount { get; set; }

        public PayRefundReq()
        {

        }

        public PayRefundReq(string merchantId, string payId, long amount)
        {
            this.MerchantId = merchantId;
            this.PayId = payId;
            this.Amount = amount;
        }

        public override string ToSign()
        {
            StringBuilder sb = new StringBuilder();
            Add(sb, MerchantId);
            Add(sb, PayId);
            Add(sb, DateTime);
            Add(sb, Amount);
            DeleteLast(sb);
            return sb.ToString();
        }
    }
}
